### PR TITLE
Use CMake from GCCcore when installing Eigen 3.3.4+

### PR DESCRIPTION
**Pull request by Micket**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/e/Eigen/Eigen-3.3.7.eb [hunk 1]
```diff
@@ -15,8 +15,13 @@ source_urls = [BITBUCKET_SOURCE]
...
 
-# stick to latest CMake 3.9.x, since more recent CMake versions require a C++ compiler that supports C++11,
-# which may not be available yet in older OSs (e.g. CentOS 6.x)
-builddependencies = [('CMake', '3.9.6')]
+# Any CMake > 2.8.5 will work. Reusing one from GCCcore from around the same time, because
+# building CMake at core level is often difficult. 
+# Change the GCCcore version if you want to reuse a different CMake you already have installed.
+builddependencies = [
+    ('GCCcore', '7.3.0'),
+    ('binutils', '2.31.1'),
```
Comment (boegel) [inline easybuild/easyconfigs/e/Eigen/Eigen-3.3.7.eb]: @Micket Please add a comment for both `GCCcore` and `binutils` to clarify why they need to be listed explicitly.

**Comment by boegel** [comment]
@Micket We should make the same change in both `Eigen-3.3.4.eb` and `Eigen-3.3..eb` as well (see #6784)?

File: easybuild/easyconfigs/e/Eigen/Eigen-3.3.7.eb (hunk 1 above)
Comment (Micket) [reply easybuild/easyconfigs/e/Eigen/Eigen-3.3.7.eb]: done

**Comment by boegel** [comment]
@migueldiascosta Please submit test report from CentOS 6.x?

@vanzod Please submit test report from non-RHEL OS?

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
node3107.skitty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 2.7.5
See https://gist.github.com/32abab2ba17ea654ad79d01a0dcc3d44 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
node2006.delcatty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/f2e941d08daa7e76d4d0027d5a2b9e69 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
node2404.golett.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/c2181aa6bf8f71c5b41cce1f710fc4d2 for a full test report.

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 5 out of 5 (3 easyconfigs in this PR)
b-an03.hpc2n.umu.se - Linux ubuntu 16.04, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 2.7.12
See https://gist.github.com/568d0c79b8c29707850f39e82ea88c1e for a full test report.

**Comment by migueldiascosta** [comment]
Test report by @migueldiascosta
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
grc-cluster1 - Linux centos 6.10, Intel(R) Xeon(R) CPU E5-2640 0 @ 2.50GHz, Python 2.7.14
See https://gist.github.com/371cc1e23cf5a89f5cfc71810124ab01 for a full test report.

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 3 out of 3 (3 easyconfigs in this PR)
hebbe-s1.int.private - Linux centos 6.10, Intel(R) Xeon(R) CPU E5-2650 v3 @ 2.30GHz, Python 2.7.14
See https://gist.github.com/506383ffd8b86bf5d935e68a0d18fbce for a full test report.

**Comment by boegel** [comment]
Going in, thanks @Micket!

**Pull request merged**